packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -833,52 +828,42 @@ export default class TabDisks extends Component {
       sr.$container === requiredHost
   )
 
-  _getVbdsByVdi = createSelector(
-    () => this.props.vdis,
+  _getVbds = createSelector(
     () => this.props.vbds,
-    () => this.props.vm,
-    (vdis, vbds, vm) => mapValues(vdis, vdi => find(vbds, { VDI: vdi.id }))
-  )
-
-  _getIsVdiAttached = createSelector(
-    createSelector(
-      () => this.props.allVbds,
-      () => Object.keys(this.props.vdis),
-      (vbds, vdis) => pick(groupBy(vbds, 'VDI'), vdis)
-    ),
-    vbdsByVdi => mapValues(vbdsByVdi, vbds => some(vbds, 'attached'))
+    () => this.props.vdis,
+    (vbds, vdis) => filter(vbds, vbd => vdis[vbd.VDI] !== undefined)
   )
 
   individualActions = [
     ...(process.env.XOA_PLAN > 1
       ? [
           {
-            handler: exportVdi,
+            handler: (vbd, { vdis }) => exportVdi(vdis[vbd.VDI]),
             icon: 'export',
             label: _('exportVdi'),
           },
           {
-            disabled: ({ id }, { isVdiAttached }) => isVdiAttached[id],
-            handler: importVdi,
+            disabled: vbd => vbd.attached,
+            handler: (vbd, { vdis }) => importVdi(vdis[vbd.VDI]),
```

Comment:
You should be able to call XO methods with the ID. Make sure to check that it's true, though.